### PR TITLE
T5050: Firewall: Add log options

### PR DESCRIPTION
**Pull request by nicolas-fort**
## Change Summary

Add log options for firewall

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T5050

## Component(s) name

firewall

## How to test

```
vyos@LATEST# compare
[]
+ firewall {
+     name FOO {
+         rule 10 {
+             action "accept"
+             log "enable"
+             log-options {
+                 level "emerg"
+             }
+             protocol "icmp"
+         }
+         rule 20 {
+             action "reject"
+             log "enable"
+             log-options {
+                 group "12"
+                 queue-threshold "64000"
+                 snapshot-length "44"
+             }
+             source {
+                 address "195.51.100.1"
+             }
+         }
+     }
+ }

[edit]
vyos@LATEST# commit
[edit]
vyos@LATEST# sudo nft list chain ip vyos_filter NAME_FOO
table ip vyos_filter {
        chain NAME_FOO {
                meta l4proto icmp log prefix "[FOO-10-A]" log level emerg counter packets 0 bytes 0 return comment "FOO-10"
                ip saddr 195.51.100.1 log prefix "[FOO-20-R]" log group 12 snaplen 44 queue-threshold 64000 counter packets 0 bytes 0 reject comment "FOO-20"
                counter packets 0 bytes 0 drop comment "FOO default-action drop"
        }
}
[edit]
vyos@LATEST# 

```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [x] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: interface-definitions/include/firewall/rule-log-options.xml.i [hunk 1]
```diff
@@ -0,0 +1,88 @@
+<!-- include start from firewall/rule-log-options.xml.i -->
+<node name="log-options">
+   <properties>
+    <help>Log options</help>
+  </properties>
+  <children>
+    <leafNode name="group">
+      <properties>
+        <help>Set log group. Log must be enable.</help>
```
Comment (c-po) [inline interface-definitions/include/firewall/rule-log-options.xml.i]: I think we can skip this redundant string "Log must be enable", it dhould be displayed during `commit` if log was not configured.

File: interface-definitions/include/firewall/rule-log-options.xml.i [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        <constraint>
+          <validator name="numeric" argument="--range 0-65535"/>
+        </constraint>
+      </properties>
+    </leafNode>
+    <leafNode name="snaplen">
+      <properties>
+        <help>Length of packet payload to include in netlink message. Log must be enable.</help>
+        <valueHelp>
+          <format>u32:0-4294967295</format>
```
Comment (c-po) [inline interface-definitions/include/firewall/rule-log-options.xml.i]: Having a packaged with length 4294967295 logged might cause a lot of trouble. I think we should limit this to 9000.

File: interface-definitions/include/firewall/rule-log-options.xml.i [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+          <description>Info log level</description>
+        </valueHelp>
+        <valueHelp>
+          <format>debug</format>
+          <description>Debug log level</description>
+        </valueHelp>
+        <constraint>
+          <regex>(emerg|alert|crit|err|warn|notice|info|debug)</regex>
+        </constraint>
+      </properties>
```
Comment (c-po) [inline interface-definitions/include/firewall/rule-log-options.xml.i:85]: Please add a `<constraintErrorMessage>` indicating only `emerg alert crit err warn notice info debug` is supported

**Review by c-po — CHANGES_REQUESTED** [review]

**Review by c-po — APPROVED** [review]

**Review by sarthurdev — APPROVED** [review]

File: interface-definitions/include/firewall/rule-log-options.xml.i [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        <valueHelp>
+          <format>u32:0-65535</format>
+          <description>Log group to send messages to</description>
+        </valueHelp>
+        <constraint>
+          <validator name="numeric" argument="--range 0-65535"/>
+        </constraint>
+      </properties>
+    </leafNode>
+    <leafNode name="snaplen">
```
Comment (sever-sever) [inline interface-definitions/include/firewall/rule-log-options.xml.i]: Should it be “snapshot-length? Instead of “snaplen”?

**Review by sever-sever — APPROVED** [review]

**Pull request merged**